platform-common-string/src/main/java/com/softicar/platform/common/string/csv/CsvTokenizer.java

```diff
@@ -0,0 +1,95 @@
+package com.softicar.platform.common.string.csv;
+
+import com.softicar.platform.common.string.CommonStringI18n;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Extracts value tokens from a CSV-formatted {@link String}.
+ * <p>
+ * Assumes a comma ({@code ,}) as value separator.
+ * <p>
+ * Assumes that escaping is done via:
+ * <ol>
+ * <li>Enclosing <i>escape-worthy</i> values in double quotes ({@code "}),
+ * and</li>
+ * <li>Duplication of any double quote that is contained in a value
+ * ({@code " -> ""})</li>
+ * </ol>
+ * A value is <i>escape-worthy</i> if it contains at least one of the following
+ * characters: {@code [,][\r][\n]["]}
+ * <p>
+ * Tolerates superfluous double quotes around values which are not
+ * <i>escape-worthy</i>.
+ * <p>
+ * Retains line breaks in values, as long as the values are quoted. This way, an
+ * extracted <i>logical row</i> can emerge from several <i>physical rows</i> in
+ * the original CSV {@link String}.
+ *
+ * @author Alexander Schmidt
+ */
```

Comment:
Did you forget to change the javadoc after you finished implementing the code?
Maybe the following text would be more suitable:

```
Assumes a comma (,) as value separator and a double quote (") as String delimiter.

A row of values must end with newline if there is another row of values. Newline must be encoded either by line feed (\n) or by carriage return and line feed (\r\n).

If a String value contains at least one of the following character sequences: "\n", "\r\n", ",", it must be enclosed by double quotes.

A double quote (") inside a String value must be represented by the escape sequence two double quotes ("").

```

Reply:
> Did you forget to change the javadoc after you finished implementing the code?

I did not. The description reflected the actual behavior of the class. I reworded it nevertheless; somewhat based upon your suggestions.